### PR TITLE
ENH: Add pandas nullable support to read_orc

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/io/test_orc.py [hunk 1]
```diff
@@ -346,3 +337,44 @@ def test_orc_use_nullable_dtypes_pyarrow_backend():
...
+            "int": pd.Series([1, 2, 3], dtype="Int64"),
+            "int_with_nan": pd.Series([1, pd.NA, 3], dtype="Int64"),
+            "na_only": pd.Series([pd.NA, pd.NA, pd.NA], dtype="Int64"),
+            "float": pd.Series([4.0, 5.0, 6.0], dtype="Float64"),
+            "float_with_nan": pd.Series([2.0, pd.NA, 3.0], dtype="Float64"),
+            "bool": pd.Series([True, False, True], dtype="boolean"),
+            "bool_with_na": pd.Series([True, False, pd.NA], dtype="boolean"),
+        }
+    )
+    tm.assert_frame_equal(result, expected)
```
Comment (gfyoung) [inline pandas/tests/io/test_orc.py:380]: Nit: could we add some newlines between some these code lines for readability

Comment (phofl) [reply pandas/tests/io/test_orc.py:386]: Done

File: pandas/io/_util.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from __future__ import annotations
+
+import pandas as pd
+
+
+def _arrow_dtype_mapping(api) -> dict:
```
Comment (mroeschke) [inline pandas/io/_util.py:6]: Might be just worth calling `import_option_dependency` here since `api` always need to be a pyarrow module?

Comment (phofl) [reply pandas/io/_util.py]: Yeah makes sense, changed

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @phofl

**Pull request merged**